### PR TITLE
[ new ] Interleaving of Lists

### PR DESCRIPTION
**Pull request by gallais**
This has been sitting on a branch for a while. Might as well start a PR.

### CODE REVIEW

File: src/Data/List/Relation/Split.agda [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+open import Relation.Binary
+open import Relation.Binary.PropositionalEquality as P using (_≡_)
+
+module _ {a b c l r} {A : Set a} {B : Set b} {C : Set c}
+         (L : REL A B l) (R : REL A C r) where
+
+  data Split : List A → REL (List B) (List C) (l ⊔ r) where
+    []   : Split [] [] []
+    _ˡ∷_ : ∀ {a b as l r} → L a b → Split as l r → Split (a ∷ as) (b ∷ l) r
+    _ʳ∷_ : ∀ {a b as l r} → R a b → Split as l r → Split (a ∷ as) l (b ∷ r)
```
Comment (MatthewDaggitt) [inline src/Data/List/Relation/Split.agda]: Is there a reason that the superscripts are on the left? We usually have them on the right...

Comment (gallais) [reply src/Data/List/Relation/Split.agda]: I don't know why I did that. In [typing with leftovers](https://gallais.github.io/typing-with-leftovers/linear.Context.Mix.html), I do have them on the right.
I'll push a patch.

**Comment by MatthewDaggitt** [comment]
Looks good! My only comment is that maybe `Split` is a bit confusing as a name as when you `split` something you typically only use a single break. `Interleave` would be much clearer. Would you object to me changing it?

**Comment by gallais** [comment]
As I said in the first commit message:

> I have changed my mind and called it `Split` because the List being broken down
is the first argument of the relation.

I am a bit uneasy because `Interleave xs ys zs` intuitively suggests to me that I get
`zs` by interleaving `xs` and `ys` rather than the other way around. That being said, it
is not that big of a deal to me so I won't oppose a name change.

**Comment by MatthewDaggitt** [comment]
> I am a bit uneasy because Interleave xs ys zs intuitively suggests to me that I get
zs by interleaving xs and ys rather than the other way around.

What about if we changed the semantics to match that intuition?

**Comment by gallais** [comment]
That could work. We would presumably need to flip the relations too to keep the
arguments in the same order as the indices'.

**Comment by MatthewDaggitt** [comment]
I'll give it a go at some point and see what it reads like.

**Comment by MatthewDaggitt** [comment]
I've updated the name to `Interleaving` and swapped the argument order. I've also thrown in `Propositional` interleavings for the sake of it, and move `toPermutation` there.

**Pull request merged**